usr/share/rear/pack/GNU/Linux/900_create_initramfs.sh

```diff
@@ -114,21 +122,73 @@ case "$REAR_INITRD_COMPRESSION" in
         else
             REAR_INITRD_FILENAME="initrd.cgz"
         fi
-
         LogPrint "Creating recovery/rescue system initramfs/initrd $REAR_INITRD_FILENAME with gzip default compression"
         if find . ! -name "*~" | cpio -H newc --create --quiet | gzip > "$TMP_DIR/$REAR_INITRD_FILENAME" ; then
             needed_seconds=$(( $( date +%s ) - start_seconds ))
-            LogPrint "Created $REAR_INITRD_FILENAME with gzip default compression ($( stat -c%s $TMP_DIR/$REAR_INITRD_FILENAME ) bytes) in $needed_seconds seconds"
+            initrd_bytes=$( stat -L -c '%s' "$TMP_DIR/$REAR_INITRD_FILENAME" )
+            initrd_MiB=$( mathlib_calculate "$initrd_bytes / 1048576" )
+            LogPrint "Created $REAR_INITRD_FILENAME with gzip default compression ($initrd_MiB MiB) in $needed_seconds seconds"
         else
             # No need to clean up things (like 'popd') because Error exits directly:
             Error "Failed to create recovery/rescue system $REAR_INITRD_FILENAME"
         fi
         ;;
 esac
+popd >/dev/null
 
 # Only root should be allowed to access the initrd
 # because the ReaR recovery system can contain secrets
 # cf. https://github.com/rear/rear/issues/3122
 test -s "$TMP_DIR/$REAR_INITRD_FILENAME" && chmod 0600 "$TMP_DIR/$REAR_INITRD_FILENAME"
 
-popd >/dev/null
+# On POWER architecture there could be an initrd size limit depending on the boot method
+# which is 128 MiB minus some MiBs for IBM's prep boot service data
+# so in practice what is left for the initrd is about 120 MiB or even less,
+# cf. the somewhat related IBM article
+# https://www.ibm.com/support/pages/system-boot-ends-grub-out-memory-oom
+# which talks about other limit values but it describes the general idea behind.
+# This article tells that one gets an "out of memory" error when the limit is exceeded
+# but in practice it also happens that the kernel starts up but fails with something like
+# "Kernel panic - not syncing: VFS: Unable to mount root fs ..."
+# cf. the 'boot.log' attachment in https://github.com/rear/rear/issues/3189
+# and then it is almost impossible to imagine that the root cause of such a kernel panic
+# is a too big initrd.
+# So to be a bit more on the safe side we at least tell the user here
+# when the initrd is bigger than 100 MiB that this may cause a boot failure:
+if test "$ARCH" = "Linux-ppc64" || test "$ARCH" = "Linux-ppc64le" ; then
```

Comment:
Is there a reason to have this stuff here and not in the appropriate subdirectory where our `SourceStage()` function would pick it up?

In general I really would like to see arch specific code to be in an arch specific directory to make reading the general scripts easier and to make them shorter.

Reply:
I have no good idea how to properly separate
arch-dependent code away into separate scripts
and still keep the code easy to understand, cf.
https://github.com/rear/rear/pull/3233#issuecomment-2145206590